### ISSUE TITLE
Build targets link incorrectly to base path compared to Serve

### ISSUE THREAD

**Issue by martinbira** [mkdocs/mkdocs#1579]
After the 1.0 upgrade all my links used as references broke when using `mkdocs build`. I've performed tests with the Default theme, ReadTheDocs theme and Material theme to rule out theme specific issues.

All my links used in .md-files are defined in the form of `[link-id]: /path/to/page/ "Title text"` and has worked as expected before the upgrade. Locally both with `mkdocs serve` and `mkodcs build` on a Win 10, and `mkdocs build` on our Linux build servers. Now when using `mkdocs build` both locally and on the build servers, all links referenced as above 404s but works correctly locally with `mkdocs serve`.

I tried to change the form of the links to `[link-id]: path/to/page/ "Title text"` but that just made things worse both with `build` and `serve`. 

I've compiled my findings below:
Local project is placed in `file:///C:/DevTest/`

### mkdocs.yml
```yaml
site_name: 'DevTest'
site_dir: 'site'
use_directory_urls: true
strict: true

nav:
  - Root:
    - Home: index.md
    - Level 1:
      - Overview: level-1/index.md
      - Level 2:
        - Source page: level-1/level-2/index.md
  - Feature 1: 
    - Home: feature-1/index.md
    - Level 1:
      - Target page: feature-1/level-1/index.md
```

### Expected behaviour
With link in .md for "Source page": `[link]: /feature-1/level-1/ "Feature 1 - Level 1"`
Expected HTML source: `<p><a href="/feature-1/level-1/" title="Feature 1 - Level 1">Feature 1 - Level 1</a></p>`
Expected serve target: `http://127.0.0.1:8000/feature-1/level-1/`
Expected build target: `file:///C:/DevTest/site/feature-1/level-1/`
Target received by right clicking the link in Google Chrome and selecting "Copy link adress".

### Actual behaviour
**Default theme**
Serve HTML source: `<p><a href="/feature-1/level-1/" title="Feature 1 - Level 1">Feature 1 - Level 1</a></p>`
Serve target: `http://127.0.0.1:8000/feature-1/level-1/`
Build HTML source: `<p><a href="/feature-1/level-1/" title="Feature 1 - Level 1">Feature 1 - Level 1</a></p>`
Build target: `file:///C:/feature-1/level-1/`

**ReadTheDocs theme**
Serve HTML source: `<p><a href="/feature-1/level-1/" title="Feature 1 - Level 1">Feature 1 - Level 1</a></p>`
Serve target: `http://127.0.0.1:8000/feature-1/level-1/`
Build HTML source: `<p><a href="/feature-1/level-1/" title="Feature 1 - Level 1">Feature 1 - Level 1</a>`
Build target: `file:///C:/feature-1/level-1/`

**Material**
Serve HTML source: `<p><a href="/feature-1/level-1/" title="Feature 1 - Level 1">Feature 1 - Level 1</a></p>`
Serve target: `http://127.0.0.1:8000/feature-1/level-1/`
Build HTML source: `<p><a href="/feature-1/level-1/" title="Feature 1 - Level 1">Feature 1 - Level 1</a></p>`
Build target: `file:///C:/feature-1/level-1/`

**Comment by waylan** [comment]
We have tightened up how internal links are processed. First, all links **must** be relative. In other words, links must not start with a slash (`/`). If you remove the slash, your links will work as expected.

Second, we have always required internal links to point to the markdown files themselves, not to the built URL. Then, when we render the Markdown, we render the URL correctly based on the config.

In past versions, if you did not do that, the URLs might still work for you. However, in the new version, this has been tightened up and you **must** use the internal URLs to ensure correct behavior.

In other words, rather than using `[link-id]: /path/to/page/ "Title text"`, you **must** use `[link-id]: path/to/page.md "Title text"`.

To be clear, this is what we have always expected you to use in your docs. The fact that it worked with any other URL scheme was a coincidence. The difference is that we not actually enforce the requirement.

**Comment by martinbira** [comment]
Got it! Time to get to som regexp then!

**Comment by martinbira** [comment]
Sorry, but I tried this with my small DevTest where as all links in the .md files are according to `[link-id]: path/to/page.md "Title text"` and relative to the root just as in the mkdocs.yml but this time the build fails.

In source page: `[link]: feature-1/level-1/index.md "Feature 1 - Level 1"`

Error message:
```
ERROR   -  Error reading page 'level-1\level-2\index.md': Documentation file 'level-1\level-2\index.md' contains a link to 'level-1\level-2\feature-1\level-1\index.md' which does not exist in the documentation directory.
Error: Documentation file 'level-1\level-2\index.md' contains a link to 'level-1\level-2\feature-1\level-1\index.md' which does not exist in the documentation directory.
```

**Comment by waylan** [comment]
Sorry, I left out the detail that all links need to be relative to the current location. This is all spelled out in the [documentation][1].

[1]: http://www.mkdocs.org/user-guide/writing-your-docs#internal-links
